src/backend/joanie/core/utils/course_run.py

```diff
@@ -0,0 +1,39 @@
+"""
+Utility for course runs on the amount of active enrollments once the course run has reached its
+end date, and the amount of orders to unlock the accessibility to get the certificate.
+"""
+import logging
+
+from django.core.exceptions import ValidationError
+from django.utils import timezone as django_timezone
+
+from joanie.core import enums
+from joanie.core.models import CourseRun, Enrollment, Order
+
+logger = logging.getLogger(__name__)
+
+
+def enrollments_by_resource_link(resource_link: str):
+    """
+    Returns a dictionary containing the following values from a `resource_link` of a course run:
+        - amount of enrollments at the end date of a course run,
+        - amount of orders made to unlock the certificates at the end of a course run.
+    """
+    if not CourseRun.objects.filter(resource_link=resource_link).exists():
+        error_message = "Make sure to give an existing resource link from a course run."
+        logger.error("Error: %s", error_message)
+        raise ValidationError(error_message)
+
+    now = django_timezone.now()
+    return {
+        "nb_enrollments_until_end": Enrollment.objects.filter(
+            course_run__resource_link=resource_link,
+            course_run__start__lte=now,
+            course_run__end__gte=now,
+            is_active=True,
+        ).count(),
+        "nb_orders_to_unlock_certificate": Order.objects.filter(
+            enrollment__course_run__resource_link=resource_link,
+            state=enums.ORDER_STATE_VALIDATED,
+        ).count(),
```

Comment:
Then you could retrieve counters like that :

```suggestion
        "nb_enrollments_until_end": Enrollment.objects.filter(
            course_run=course_run,
            is_active=True,
        ).count(),
        "nb_orders_to_unlock_certificate": Order.objects.filter(
            enrollment__course_run=course_run,
            product__type=enums.PRODUCT_TYPE_CERTIFICATE,
            state=enums.ORDER_STATE_VALIDATED,
        ).count(),
```

To add robustness to your order query, you could also filter by `product__type`.

Reply:
Meanwhile we fix the following [issue](https://github.com/openfun/joanie/issues/529), I have added another filter in the `nb_orders_to_unlock_certificate` queryset such as : 

```python
"nb_orders_to_unlock_certificate": Order.objects.filter(
    enrollment__course_run=course_run,
    enrollment__course_run__end__lte=django_timezone.now(),
    product__type=enums.PRODUCT_TYPE_CERTIFICATE,
    state=enums.ORDER_STATE_VALIDATED,
).count(),
```